### PR TITLE
Add Hide/Show Completed Items Feature for Notes

### PR DESCRIPTION
**Pull request by igennova**
Enhancement: #411 
- **Hide/Show Toggle**: Added eye/eye-off button to note headers for hiding completed checklist items
- **Per-Note Control**: Each note maintains its own hide/show state independently  
- **Smart Visibility**: Toggle only appears when note has completed items to hide
- **Clean UI**: Hides completed tasks to help users focus on remaining work
- **Comprehensive Testing**: Added 3 E2E tests covering hide/show functionality and edge cases
Video

https://github.com/user-attachments/assets/3d4ab90f-d3ee-414f-8643-831acc907188

Test Case:
image

### CODE REVIEW

File: components/note.tsx [hunk 1]
```diff
@@ -376,6 +381,28 @@ export function Note({
           </div>
         </div>
         <div className="flex items-center space-x-2">
+          {/* Hide/Show Completed Toggle - only show if there are completed items */}
```
Comment (slavingia) [inline components/note.tsx:384]: Can remove all these kinds of comments

File: components/note.tsx [hunk 2]
```diff
@@ -471,15 +498,15 @@ export function Note({
         <div className="overflow-y-auto space-y-1">
           {/* Checklist Items */}
           <DraggableRoot
-            items={note.checklistItems ?? []}
```
[the remark this answers is not in the document]
Comment (slavingia) [reply components/note.tsx:474]: What happens once you expand again?

**Comment by slavingia** [comment]
I think a settings to auto sort all completed at bottom may be better than this - it’s a lot of real estate for a rarely used feature
